src/main/java/com/github/bot/curiosone/core/extraction/Conversation.java

```diff
@@ -18,15 +18,20 @@
 import java.util.stream.Stream;
 
 /**
- * Precomputed answers to some common conversational phrases.
+ * Handles the precomputed answers to some common conversational phrases.
+ * Provides an utility method to get an answer from a given sentence.
  */
 public class Conversation {
-  private static String conversationsPath = "/conversation/conversation.txt";
-  /** Path of the database containing known answers. */
 
+  /**
+   * Path to the database containing known answers.
+   */
+  private static String conversationsPath = "/conversation/conversation.txt";
 
-  /** Map from recognized tokens to possible phrases given in output. */
-  private static LinkedHashMap<String, String[]> knownQuestions;
+  /**
+   * Maps the recognized tokens to their possible phrases.
+   */
+  private static Map<String[], String[]> knownQuestions;
```

Comment:
You should keep the LinkedHashMap.
We need to store the keys in the reading order from the file.